Xamarin.PropertyEditing.Mac/Controls/Custom/ResourceBrushViewController.cs

```diff
@@ -0,0 +1,235 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using AppKit;
+using CoreGraphics;
+using Foundation;
+using Xamarin.PropertyEditing.Drawing;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	class ResourceDataSource : NSOutlineViewDataSource
+	{
+		public ResourceDataSource (ResourceSelectorViewModel viewModel) : base ()
+		{
+			this.vm = viewModel;
+		}
+
+		public override nint GetChildrenCount (NSOutlineView outlineView, NSObject item)
+		{
+			if (this.vm?.Resources == null)
+				return 0;
+
+			if (this.vm.Resources?.Count () == 0)
+				return 0;
+
+			return this.vm.Resources.Count ();
+		}
+
+		public override NSObject GetChild (NSOutlineView outlineView, nint childIndex, NSObject item)
+		{
+			object element;
+
+			element = (this.vm.Resources.ElementAt ((int)childIndex));
```

Comment:
Really I don't think there's any reason we can't move `SimpleCollectionView` to implement `IList` instead of doing this, there could potentially be a lot of resources that this is iterating over every time, it's not going to hit a fast path currently.

Reply:
I'll fix this as soon as #339 lands